### PR TITLE
Wiki issue 62

### LINKED ISSUE #62: Document giraf_app_bar_widget.dart
The overall purpose of documenting the program is to describe the overall structure of the program and not specifically code.

For solving this issue you can do the following:

Relations:
- Write about which BLoCs/Screens this class is related to.

Responsibility/Functionality:
- Write about the responsibility/functionality of the current class and how it interacts with it’s related BLoCs/Screens (e.g. this stream is used to send weekplans to this screen, so they can be displayed).

Note: If you have any suggestions to what we can change, do not hesitate to contact us :D

### PR DESCRIPTION
**Pull request by User12837985123**
This closes #62.

### CODE REVIEW

**Comment by User12837985123** [comment]
@aau-giraf/sw608f19 you have been assigned this. Here's your checklist. I marked @swuf as the reviewer, but you might change it as you like.
Please use the following checklist:
- [ ] Is there documented which BLoCs/Screens the current class is related to?

- [ ] Is the responsibility/functionality of the current class explained? Also regarding the classes it is related to?

- [ ] If any unusual decisions are introduced in the code are they reasoned about in the documentation?

File: file_documentation/weekplanner/widgets/giraf_app_bar_widget.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Documentation of giraf_app_bar_widget.dart
+This files contains the documentaion of the `giraf_app_bar_widget.dart` file from the weekplanner project.
+
+## Functionality
+This widget is the appbar at the top of many screens in the weekplanner application. The displayed icons are chosen dynamically and depends on the current screen and user mode. The icons as well as their functionality are provided by `toolbar_bloc.dart`. 
+
+## Layout
+The Giraf appbar is displayed as seen here, when located on the weekplan screen (`weekplan_screen.dart`):
+
+![Layout of Giraf app bar](../pictures/GirafAppbar.PNG)
```
Comment (swuf) [inline file_documentation/weekplanner/widgets/giraf_app_bar_widget.md]: Image doesn't seem to work. Maybe it's the .PNG with capital letters.

File: file_documentation/weekplanner/widgets/giraf_app_bar_widget.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Documentation of giraf_app_bar_widget.dart
+This files contains the documentaion of the `giraf_app_bar_widget.dart` file from the weekplanner project.
+
+## Functionality
+This widget is the appbar at the top of many screens in the weekplanner application. The displayed icons are chosen dynamically and depends on the current screen and user mode. The icons as well as their functionality are provided by `toolbar_bloc.dart`.
```
Comment (swuf) [inline file_documentation/weekplanner/widgets/giraf_app_bar_widget.md]:
```suggestion
This widget is the app bar at the top of many screens in the weekplanner application. The displayed icons are chosen dynamically and depend on the current screen and user mode. The icons as well as their functionality are provided by `toolbar_bloc.dart`. 
```

File: file_documentation/weekplanner/widgets/giraf_app_bar_widget.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Functionality
+This widget is the appbar at the top of many screens in the weekplanner application. The displayed icons are chosen dynamically and depends on the current screen and user mode. The icons as well as their functionality are provided by `toolbar_bloc.dart`. 
+
+## Layout
+The Giraf appbar is displayed as seen here, when located on the weekplan screen (`weekplan_screen.dart`):
+
+![Layout of Giraf app bar](../pictures/GirafAppbar.PNG)
+
+## Buttons
+The appbar can contain many different buttons. To name a few: settings, logout, change to guardian, and change to citizen, all the icons can bee seen in `toolbar_bloc.dart`.
```
Comment (swuf) [inline file_documentation/weekplanner/widgets/giraf_app_bar_widget.md]:
```suggestion
The app bar can contain many different buttons. To name a few: settings, logout, change to guardian, and change to citizen. All the icons can be seen in `toolbar_bloc.dart`.
```

File: file_documentation/weekplanner/widgets/giraf_app_bar_widget.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## Code
+Giraf appbar is a stateless widget, thus it contains the build function which "describes" how the appbar should look:
```
Comment (swuf) [inline file_documentation/weekplanner/widgets/giraf_app_bar_widget.md]:
```suggestion
Giraf app bar is a stateless widget, thus it contains the build function which "describes" how the app bar should look:
```

File: file_documentation/weekplanner/widgets/giraf_app_bar_widget.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                return Row(
+                  children: snapshot.data
+                );
+              }),
+        ],
+        automaticallyImplyLeading: isGuardian,
+    );
+  }
+  ```
+The Giraf appbar consists of an appbar with buttons which are emitted from the stream called `toolbarBloc.visibleButtons`. That stream is populated with buttons by the method `toolbarBloc.updateIcons` which is the reason it is called as the first method.
```
Comment (swuf) [inline file_documentation/weekplanner/widgets/giraf_app_bar_widget.md]:
```suggestion
The Giraf app bar consists of an app bar with buttons which are emitted from the stream called `toolbarBloc.visibleButtons`. That stream is populated with buttons by the method `toolbarBloc.updateIcons` which is the reason it is called as the first method.
```

**Review by swuf — CHANGES_REQUESTED** [review]

**Review by swuf — APPROVED** [review]
Nice 🐤

**Pull request merged**